include/swift/Basic/Statistic.h

```diff
@@ -13,7 +13,10 @@
 #ifndef SWIFT_BASIC_STATISTIC_H
 #define SWIFT_BASIC_STATISTIC_H
 
+#include "llvm/ADT/SmallString.h"
 #include "llvm/ADT/Statistic.h"
+#include "swift/Basic/Timer.h"
+#include "swift/FrontendTool/FrontendTool.h"
```

Comment:
Hm. Driver and FrontendTool both import Frontend, but neither imports the other, and they certainly can't *both* import each other. In your case you might be able to get away with forward declarations for the reference types, but the dependency kind won't work.

Really I'm unhappy with the recording code living inside the stats mechanism at all. The end result is that the implementation file ends up with tons of references to the internal structure of several different subsystems. It's weird to see the Driver's DependencyKind mixed in with counting various SIL entities. I'd prefer if it were inverted and the various members incremented directly.